blockchain.py

```diff
@@ -49,7 +49,7 @@ def valid_chain(self, chain):
                 return False
 
             # Check that the Proof of Work is correct
-            if not self.valid_proof(last_block['proof'], block['proof']):
+            if not self.valid_proof(last_block['proof'], block['proof'], last_block['previous_hash']):
```

Comment:
Shouldn't **last_block**['previous_hash'] simply be **block**['previous_hash'] ?

Otherwise I get an error with the consensus mechanism: 
Making two test-miners and doing a /nodes/resolve on the one with a shorter chain will not work

Reply:
Thanks. It looks like this is the same bug @2gotgrossman had and fixed in line 162, but we forgot to fix it here in valid_chain.